corehq/apps/sms/handlers/form_session.py

```diff
@@ -166,7 +166,7 @@ def validate_answer(event, text, v):
     # Validate longint
     elif event.datatype == "longint":
         try:
-            long(text)
+            int(text)
```

Comment:
I'm not confident in this one - the datatype explicitly states `longint` for some reason (it's not clear to me why without context).  We might need to examine more closely if we plan to abolish the distinction between `int`s and `long`s.

Reply:
might be reasonable to use `(long if six.PY2 else int)` here?

Reply:
I think ```int``` suffices:

python2:
```
>>> int('9223372036854775807')
9223372036854775807
>>> int('9223372036854775808')
9223372036854775808L
```

python3:
```
>>> int('9223372036854775807')
9223372036854775807
>>> int('9223372036854775808')
9223372036854775808
```

That said, I wonder if ```validate_answer``` is correct to succeed as-is for the text input ```'9223372036854775808'``` when ```event.datatype == "int"```. @gcapalbo thoughts?

Reply:
Although, this would fail in both python 2 and 3:
```
>>> int('1L')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '1L'
```

Whereas in python 2:
```
>>> long('1L')
1L
```

Reply:
for the purposes of this PR, it's ok to just use `int()` here. what this is referring to is when people answer sms surveys (so filling out a form) through touchforms, and it's doing a first level of validation on their input.

but there are a couple things here that i'll clean up - it seems that the form builder has actually done away with the long datatype, and also as Nick pointed out the behavior of int() is not actually restricting to 32 bits as the [integer question type will](https://confluence.dimagi.com/display/commcarepublic/Form+Builder#FormBuilder-3.NumericQuestions), so the integer type validation has been wrong.

i've got a note to myself to clean those parts up, but in the meantime this is fine as is since it will behave the same as before